java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java

```diff
@@ -68,7 +68,9 @@ public class BarrageSnapshotImpl extends ReferenceCountedLivenessNode implements
     private volatile int connected = 1;
     private static final AtomicIntegerFieldUpdater<BarrageSnapshotImpl> CONNECTED_UPDATER =
             AtomicIntegerFieldUpdater.newUpdater(BarrageSnapshotImpl.class, "connected");
-
+    private volatile int completed = 0;
```

Comment:
No need to bother changing this, since you're planning to converge these impls in the next issue, but it would be better if the parallel variables between snapshot and sub were declared in the same order, etc.